OpenRA.Game/Map/MapCache.cs

```diff
@@ -30,16 +30,14 @@ public sealed class MapCache : IEnumerable<MapPreview>, IDisposable
 		readonly Dictionary<IReadOnlyPackage, MapClassification> mapLocations = [];
 		public bool LoadPreviewImages = true;
 
-		readonly Cache<string, MapPreview> previews;
+		readonly ConcurrentCache<string, MapPreview> previews;
```

Comment:
This by itself won't be enough to fix a race condition with remote map updates.

Consider what happens when the player switches into the MP browser before user maps have been loaded. The server list will trigger a RC search for all unknown maps, including ones that the user does actually have. As each local map is parsed it will overwrite the map preview from remote to local without checks (good). The problem is then when the RC query returns (either with data or a not found) and the map preview is again updated withotu checks changing from local to remote or invalid (bad!).

This is what the "Reorganise MapPreview.UpdateRemoteSearch" commit in #21855 aims to solve.